add example dev workflow (#6986)
Rdatatable/data.table: .dev/README.md

# data.table developer

> Inside this repository we provide some tools that improves development experience. Most notable one is `cc()` helper function that recompiles C sources, reload R sources and run tests.  

Reviewer: ```suggestion
Inside this repository we provide some tools to improves development experience. Most notable is the `cc()` helper function that recompiles C sources, reloads R sources, and runs tests.
```

Typical development workflow will then look like:

0. `git checkout -b [branch]`
1. edit package files
2. run `R`
3. call `cc(TRUE)`
4. (if needed) go to point 1.

Once we (and tests) are satisfied with changes, we then run complete package checks:

0. in shell terminal
1. run `make build`
2. run `make check`

> 3. (optionally) run `R=~/build/R-devel/bin/R make check`
> 4. (optionally) run `R=~/build/R-340/bin/R make check`

Reviewer: ```suggestion
3. (optionally) run on `r-devel`, e.g. `R=~/build/R-devel/bin/R make check`
4. (optionally) run on ancient R, e.g. `R=~/build/R-340/bin/R make check`
```
